src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -582,6 +588,68 @@ let create_fsevents ?(latency = 0.2) ~(scheduler : Scheduler.t) () =
   ; sync_table
   }
 
+let fswatch_win_exclude_patterns =
+  List.map ~f:Re.compile
+    [ Re.str "/_opam"
+    ; Re.str "/_esy"
+    ; Re.str
+        "/.#" (* Such files can be created by Emacs and also Dune itself. *)
+    ; Re.seq [ Re.char '~'; Re.eos ]
+    ; Re.str "/#"
+    ; Re.str "/4913" (* https://github.com/neovim/neovim/issues/3460 *)
+    ; Re.str "/.git"
+    ; Re.str "/.hg"
+    ; Re.seq [ Re.bos; Re.any; Re.str ":/windows" ]
```

Comment:
can we augment the existing `exclude_patterns` instead of duplicating most of it here?

Reply:
We could; but it is a bit ugly because we need the textual regexps (as opposed to `Re.t` values) for the `fswatch` fallback.

By the way, if we have native backends for the three main architectures (Linux, Windows, OS X), do we still need the `fswatch` fallback?

Reply:
If we have a list for all the first class backends and another list for fswatch that would be fine by me.

Reply:
> We could; but it is a bit ugly because we need the textual regexps (as opposed to `Re.t` values) for the `fswatch` fallback.
> 
> By the way, if we have native backends for the three main architectures (Linux, Windows, OS X), do we still need the `fswatch` fallback?

Since the Diskuv OCaml uses a `fswatch` shim on Windows, could we wait at least one release cycle before removing it?

Reply:
We still need it for BSD. It's not going anywhere.

Reply:
OK, I pushed a commit to shar the regexps everywhere.